### ISSUE TITLE
Thermal Runaway after few minutes in the print

### ISSUE THREAD

**Issue by ffredyk** [mriscoc/Ender3V2S1#841]
### Did you test with a precompiled firmware?

Yes, and the problem still exists.

### Bug Description

After a minute into a print, temperature of the nozzle starts falling rapidly. I have completed MPC Autotune (and stored) and I have tried to change various MPC parameters, but the result is always the same (only change is the start of the fall delayed or slowed)

I am using BLTouch firmware: [Ender3V2-422-BLTUBL-MPC-20230312.bin](https://github.com/mriscoc/Ender3V2S1/releases/download/20230312/Ender3V2-422-BLTUBL-MPC-20230312.bin)
G-code is generated in PrusaSlicer with changed Prefix and Postfix g-code from [wiki in this repo](https://github.com/mriscoc/Ender3V2S1/wiki/Slicer-G-code-Scripts) - but the problem persists even with PrusaSlicer's default prefix/postfix code
G-code example used: [Gist link](https://gist.github.com/ffredyk/78967d875011bcd04287eabb4570e1a1)

I am printing PETG at 240C. Intro lines are ok but when the printer starts printing the outer skirt, temperature starts to fall.
It falls to 230C and thermal runaway triggers, restarting the FW.
If I disable thermal runaway detection, then temperature continues to fall to 180C at which point extruder is not able to push any more filament into hotend and I cancelled the print.
The fall itself is not linear, but more step-like.. It begins slow, accelerates to about 1C per second and then slows again when it falls by 10C - it doesn't stop, it slowly accelerates again and repeats. 

When I heat up the nozzle without active printing and manually push filament through extruder, temperature is stable. When I use the purge function in filament changing settings, temperature still stays stable. It starts to fall only when the head is moving while printing.

Is this a settings problem? Or HW fault in PSU? How can I proceed to fix this?

### Bug Timeline

now

### Expected behavior

Nozzle temperature to stay stable while printing

### Actual behavior

Nozzle temperature falls until "Thermal runaway" error stops and restarts the printer - or when disabled - temperature drops below melting point of filament and stops extruder, failing print as the result.

### Version of Professional Firmware

M115

### Printer model

Ender 3V2

### Electronics

Board 4.2.2

### Add-ons

BLTouch

### Bed Leveling

UBL Bilinear mesh

### Your Slicer

Prusa Slicer

### Host Software

SD Card (headless)

### Additional information & file uploads

G-code: https://gist.github.com/ffredyk/78967d875011bcd04287eabb4570e1a1
I will record video in a few hours

**Comment by mriscoc** [comment]
Ender3V2-422-BLTUBL-MPC-20230312.bin was tested with stock printer with CR/BLTouch without any issue.

> ### Did you test with a precompiled firmware?
> Yes, and the problem still exists.
> 
> ### Bug Description
> After a minute into a print, temperature of the nozzle starts falling rapidly. I have completed MPC Autotune (and stored) and I have tried to change various MPC parameters, but the result is always the same (only change is the start of the fall delayed or slowed)
> 
> I am using BLTouch firmware: [Ender3V2-422-BLTUBL-MPC-20230312.bin](https://github.com/mriscoc/Ender3V2S1/releases/download/20230312/Ender3V2-422-BLTUBL-MPC-20230312.bin) G-code is generated in PrusaSlicer with changed Prefix and Postfix g-code from [wiki in this repo](https://github.com/mriscoc/Ender3V2S1/wiki/Slicer-G-code-Scripts) - but the problem persists even with PrusaSlicer's default prefix/postfix code G-code example used: [Gist link](https://gist.github.com/ffredyk/78967d875011bcd04287eabb4570e1a1)
> 
> I am printing PETG at 240C. Intro lines are ok but when the printer starts printing the outer skirt, temperature starts to fall. It falls to 230C and thermal runaway triggers, restarting the FW. If I disable thermal runaway detection, then temperature continues to fall to 180C at which point extruder is not able to push any more filament into hotend and I cancelled the print. The fall itself is not linear, but more step-like.. It begins slow, accelerates to about 1C per second and then slows again when it falls by 10C - it doesn't stop, it slowly accelerates again and repeats.
> 
> When I heat up the nozzle without active printing and manually push filament through extruder, temperature is stable. When I use the purge function in filament changing settings, temperature still stays stable. It starts to fall only when the head is moving while printing.
> 
> Is this a settings problem? Or HW fault in PSU? How can I proceed to fix this?
> 
> ### Bug Timeline
> now
> 
> ### Expected behavior
> Nozzle temperature to stay stable while printing
> 
> ### Actual behavior
> Nozzle temperature falls until "Thermal runaway" error stops and restarts the printer - or when disabled - temperature drops below melting point of filament and stops extruder, failing print as the result.
> 
> ### Steps to Reproduce
> _No response_
> 
> ### Version of Professional Firmware
> M115
> 
> ### Printer model
> Ender 3V2
> 
> ### Electronics
> Board 4.2.2
> 
> ### Add-ons
> BLTouch
> 
> ### Bed Leveling
> UBL Bilinear mesh
> 
> ### Your Slicer
> Prusa Slicer
> 
> ### Host Software
> SD Card (headless)
> 
> ### Additional information & file uploads
> G-code: https://gist.github.com/ffredyk/78967d875011bcd04287eabb4570e1a1 I will record video in a few hours

**Comment by ffredyk** [comment]
> Ender3V2-422-BLTUBL-MPC-20230312.bin was tested with stock printer with CR/BLTouch without any issue.

I am sure it was tested, my printer is also stock version, with only the BLTouch mod installed. I created the issue to find out if this is a local problem only on my machine (and some vectors to investigate and potentially find the source of the problem), or for other people to contribute if they have a similar problem (there is some reddit/forums heat surrounding this problem on the internet). The printer worked fine on stock firmware, this issue came with jyers FW and yours FW (since jyers is stale, I created an issue here)

**Comment by mriscoc** [comment]
Do you have the heat block silicon jacket in place?

Some stock and old firmwares don't have the thermal protection activated.

**Comment by Bucko-m** [comment]
I had similar problem I replaced the thermal sensor no problems since.

**Comment by LouMinartey** [comment]
DISREGARD - my thermistor was trying to leave the housing of the heat block, derp

**Comment by ffredyk** [comment]
UPDATE - So I purchased a new original hotend and replaced the prolematic one - problem solved. 

After further investigation and experimentation, I figured there was a problem with cable interface on the heating element in the block. After moving the cables slightly, the heating element started working, and then by moving the cables again, stopped heating again. The temperature sensor was fine, the heating element itself got nuked.

I hope, this info can help anyone who stumbles upon this issue by any way. 
And I am sorry @mriscoc it was indeed a false alarm :(
